index.php

```diff
@@ -0,0 +1,43 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Library of functions and constants for module observation
+ *
+ * @package mod_observation
+ * @copyright  2021 Endurer Solutions Team
+ * @author Matthew Hilton <mj.hilton@outlook.com>
+ * @license    http://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+require_once('../../config.php');
+
+global $CFG;
+global $DB;
+
+// Course id.
+$id = required_param('id', PARAM_INT);
+
+// Ensure that the course specified is valid.
+if (!$course = $DB->get_record('course', array('id' => $id))) {
```

Comment:
I dont know that this check is required. If the course is non-valid, the require_login will likely throw the same error. Test + confirm that you get an error with just require_login, if so, kill this

Reply:
Require login does not check if the course is invalid. I believe this is because if the course doesn't exist, the record returned will be NULL, and if you pass a null course to the require_login, all it does is check login and not logged into a particular course.